discovery/syncer.go

```diff
@@ -1017,13 +1045,14 @@ func (g *GossipSyncer) replyChanRangeQuery(query *lnwire.QueryChannelRange) erro
 			return err
 		}
 
+		prevChannelChunk = channelChunk
```

Comment:
Assuming most new nodes are now upgraded, I think we can also observe the "proper" semantics of the `Complete` filed if we want to here. So we'll just always et it to `1` and remove all other checks for it, since the "official" usage itself was never very well defined. Just a bit of commentary, don't think we need to make a change in this area critically.

Reply:
Good idea, we should be able to remove the legacy handling of queries. Won't include it in this PR however as there's enough going on already.